controller/server.go

```diff
@@ -29,6 +29,19 @@ func (rw *ReadWrite) Task(ctx context.Context, taskName string) (config.TaskConf
 	return config.TaskConfig{}, fmt.Errorf("a task with name '%s' does not exist or has not been initialized yet", taskName)
 }
 
+func (rw *ReadWrite) Tasks(ctx context.Context) ([]config.TaskConfig, error) {
+	// TODO handle ctx while waiting for state lock if it is currently active
+	tasks := rw.state.GetAllTasks()
+
+	// convert config.TaskConfigs => []config.TaskConfig
```

Comment:
Would it make sense in the future to have `Tasks()` return `config.TaskConfigs`? I haven't looked into where this all is called, so not sure the impact of this potential change.

Reply:
Thanks for thinking of this! I gave it a try and it trickles down some minor changes mostly in the api package. Since it also involves updating the server interface and regenerating the mock, I'd like to make this change at the same time as updating Tasks() to no longer return an error ([earlier comment](https://github.com/hashicorp/consul-terraform-sync/pull/799#discussion_r849699691)) if that works for you?

Reply:
Thanks for thinking of this! I tried it out and it was mostly minor trickle down changes in the `api` package.

I'd like to make this change together later when I update Tasks() to remove returning an error ([earlier comment](https://github.com/hashicorp/consul-terraform-sync/pull/799#discussion_r849699691)) since they similarly update the return signature of `Tasks()` if that works for you?

Reply:
That sounds good to me!

Reply:
Following up: this is addressed here https://github.com/hashicorp/consul-terraform-sync/pull/846/commits/2931a2a135d1471bd16a1735af21d50812aae9a9